src/elexmodel/utils/math_utils.py

```diff
@@ -93,3 +93,20 @@ def compute_mean_pi_length(lower, upper, pred):
     """
     # we add 1 since pred can be literally zero
     return np.mean((upper - lower) / (pred + 1)).round(decimals=2)
+
+
+def get_kfold_splits(data, n_splits):
+    splits = []
+    num_samples = len(data)
+    fold_size = num_samples // n_splits
+    remaining_samples = num_samples % n_splits
+    current_index = 0
+
+    for i in range(n_splits):
+        fold_length = fold_size + i if i < remaining_samples else fold_size
+        test_index = range(current_index, current_index + fold_length)
+        train_index = list(set(range(num_samples)) - set(test_index))
```

Comment:
I think this may need to be `range(num_samples + 1)` but a unit tests should clear that up should clear that up

Reply:
Based on my unit tests, I left it range(num_samples) but let me know what you think!

Reply:
When I pass `data = [1, 2, 3, 4, 5, 6, 7, 8, 9]` and `n_splits=4` then I get this:
```
[([2, 3, 4, 5, 6, 7, 8], range(0, 2)), ([0, 1, 4, 5, 6, 7, 8], range(2, 4)), ([0, 1, 2, 3, 6, 7, 8], range(4, 6)), ([0, 1, 2, 3, 4, 5, 8], range(6, 8))]
```
but subsetting is not inclusive, so if I grab `[2:8]` from `data` I end up with `[3, 4, 5, 6, 7, 8]` instead of `[3, 4, 5, 6, 7, 8, 9]`.  Also I think we'll wat to cast those ranges into lists explicitly (so on line 120) just to be safe.

Reply:
I'm not 100% sure what the conclusion of this is

Reply:
Just added a commit that should fix this

Reply:
Since we are using iloc, I don't think we lose that last number- I added a unit test with this example